### DISCUSSION TITLE
Yourls changes original URLs

### DISCUSSION THREAD

**Discussion by opowoj** [YOURLS/YOURLS#3184] [Q&A] [answered]
My YOURLS installation yesterday picks out all blanks in the original URL before shortening it. Therefore I get a "not found" message. How can I avoid this feature??

**Comment by dgw** [comment, marked as answer]
What is a blank? Any characters that are not allowed in a URL (including whitespace like the humble `%20` space character) [must be escaped](https://stackoverflow.com/a/1547940/5991).

**Comment by opowoj** [comment]
Hi dgw,
thanx a lot for your answer. I'm not a native speaker: blank = space = whitespace. Plenty of my (media) files have names with spaces. If I call them with Firefox via en URL from my server, like 

**https://retslop-subdom5a.synology.me/wp-content/uploads/test/Das Geheimnis der Meister - Van Gogh - Film von Jaap van der Werff und Finbarr Wilbrink.ts**

it works fine. But I understand: If i will use the URL via YOURLS, I need to change the original filename with some regex tool (i.e. RegexRenamer) before

I hoped, I can avoid this with some setting in the YOURLS dashboard or somewhere else

**Reply by dgw** [reply]
You see that GitHub also doesn't recognize the rest of the link after a space? 😁

No need to rename the original files. The exact trick will depend on how you're shortening the links, but if you search for a way to "urlencode" before pasting the link into YOURLS you'll be on the right track.

Many browsers accept and display the unencoded URL to avoid scaring users, but internally I guarantee you that Firefox is converting your spaces into `%20` so the server will understand it.

**Reply by opowoj** [reply]
> but if you search for a way to "urlencode" before pasting the link into YOURLS you'll be on the right track.

thank you so much for the 2nd answer. I will "urlencode" and be on the right track in the future
